File: backend/secondary/campus-backend/src/main/java/com/campus/backend/model/Presence.java
package com.campus.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "presence")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Presence {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    private PresenceStatus status = PresenceStatus.OFFLINE;

    private String location;

    private LocalDateTime lastSeen = LocalDateTime.now();

    @Column(updatable = false)
    private LocalDateTime checkIn = LocalDateTime.now();

    private LocalDateTime checkOut;
}

File: mobile/Campus/app/src/main/java/com/campus/app/data/model/Presence.kt
@file:OptIn(kotlinx.serialization.InternalSerializationApi::class)

package com.campus.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Presence(
    val userId: Long,
    val userName: String,
    val status: String,
    val location: String?,
    val lastSeen: String
)

@Serializable
data class PresenceUpdateRequest(
    val status: String,
    val location: String?
)